packages/core/strapi/lib/services/entity-service/types/result.d.ts

```diff
@@ -2,24 +2,63 @@ import type { Attribute, Common, EntityService, Utils } from '@strapi/strapi';
 
 type Pagination = { page: number; pageSize: number; pageCount: number; total: number };
 
-export type Entity<TSchemaUID extends Common.UID.Schema> = Utils.Object.DeepPartial<
-  GetValues<TSchemaUID>
+export type Entity<
```

Comment:
Without types defined, both examples throw an error because of the `bar` property

```ts
const entity: Entity<'api::foo.foo', { fields: 'bar' }> = {
  id: 'foo',
  bar: 'foo', // invalid prop
};

const entity: Entity<'api::foo.foo', never> = {
  id: 'foo',
  bar: 'foo', // invalid prop
};
```

We should probably have a default value for the params (can be never or anything else) + do something else if the schema related to the UID doesn't exists (types not defined for example). For example we could return a mapped type like `{ [x: string]: unknown/any }.`

What do you think?